airbyte-db/lib/configs_db_schema.md

```diff
@@ -0,0 +1,359 @@
+  - Will we ever need non-standard models like custom workspace or custom connector definition?
+- Most source and destination tables have exactly the same columns. For example, `SOURCE_DEFINITION` and `DESTINATION_DEFINITION`, `SOURCE_OAUTH_PARAM` and `DESTINATION_OAUTH_PARAM`. Should we merge these tables, and add a `type` column to distinguish the rows?
+  - Things to consider
+    - Dry-ness: one table is better
+    - Query complexity: same
+    - Add new columns
+      - Separate tables: need to add new columns to both tables.
+      - One table: add new columns to one table.
+      - One table is better.
+  - Do we expect the source and definition models to diverge in the future?
+    - If we use one table for both source and definition models, and they diverge in the future, we can use the table-per-type strategy and simply add extra tables for the model that needs extra properties. Still one table is better
+  - Based on the above dimensions, I am leaning towards merging the two tables.
+
+## Common Columns
+| Column | Type  | Required | Default | Notes |
+| ------ | :---: | :---:    | :---:   | ---   |
+| id | char(36) | yes | | Primary key, UUID. |
+| created_at | timestamp with time zone | yes | current_timestamp | `current_timestamp` is implemented in most databases. |
+| updated_at | timestamp with time zone | yes | current_timestamp | `current_timestamp` is implemented in most databases. |
+
+## Tables
+
+### `STANDARD_WORKSPACE`
+
+YAML [reference](https://github.com/airbytehq/airbyte/blob/master/airbyte-config/models/src/main/resources/types/StandardWorkspace.yaml)
+
+| Column | Type  | Required | Default | Notes |
+| ------ | :---: | :---:    | :---:   | ---   |
+| customer_id | char(36) | | | Not a foreign key. |
+| name | varchar(255) | yes | | |
+| slug | varchar(255) | yes | | |
+| email | varchar(255) | | | index |
+| initial_setup_complete | boolean | yes | false | |
+| anonymous_data_collection | boolean | | true | |
+| news | boolean | | false | | |
+| security_updates | boolean | | false | |
+| display_setup_wizard | boolean | | true | Artem suggested to default this field to `true`. |
+| tombstone | boolean | | false | index |
+
+### `WORKSPACE_NOTIFICATION`
+
+YAML reference:
+- [Notification](https://github.com/airbytehq/airbyte/blob/master/airbyte-config/models/src/main/resources/types/Notification.yaml)
+
+This table has many-to-one relationship with the `WORKSPACE` table. One workspace can have multiple notifications.
+
+This is the base table for notifications. Each notification type will have its own child table (e.g. `SLACK_NOTIFICATION`).
+
+| Column | Type  | Required | Default | Notes |
+| ------ | :---: | :---:    | :---:   | ---   |
+| workspace_id | char(36) | yes | | Foreign key. Index. |
+| notification_type | varchar(50) | yes | | Index. String enum: "slack". |
+| send_on_success | boolean | | false | index |
+| send_on_failure | boolean | | false | index |
+
+### `SLACK_WORKSPACE_NOTIFICATION`
```

Comment:
it's not super obvious to me that we get a lot of value in having this be a separate table. throwing this as a json blob in the `workspace_notification` table would be fine. I say this because as we add more config types we're going to have to have a db table per integration which i think might be a headache (e.g. slack, webhook, pagerduty, email, etc, this list could get pretty long). I also don't think we would ever really want to query on the contents of this data.